### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by loic**
- Move couchdb-harness to devDependencies.
- Remove duplicated pouchdb-rewrite tests.
- Removed pouchdb-seamless-auth dep.

### CODE REVIEW

File: tests/pouchdb-rewrite/features.js [hunk 1]
```diff
@@ -248,15 +248,15 @@ describe('sync rewrite tests', () => {
     })
 
     .then((error) => {
-      error.status.should.equal(401);
+      error.status.should.equal(404);
```
Comment (loic) [inline tests/pouchdb-rewrite/features.js]: @marten-de-vries would you know why this change and the one line 259 are needed when removing `pouchdb-seemsless-auth`? Is that concerning? (I haven't had time to investigate).

**Comment by marten-de-vries** [comment]
Any idea why the rewrite tests were in there double in the first place?

File: tests/pouchdb-rewrite/features.js (hunk 1 above)
Comment (marten-de-vries) [reply tests/pouchdb-rewrite/features.js]: Ah, so this was why ``pouchdb-seamless-auth`` was included. The changed test is essentially worthless. The test essentially tests parts of ``pouchdb-route`` and ``pouchdb-seamless-auth``. Both are relatively useless from an ``express-pouchdb`` perspective, but the first is essential for maintaining the current ``pouchdb-rewrite`` API.

**Comment by marten-de-vries** [comment]
So the problem here is whether to include projects in the monorepo that are for test suite use only: ``pouchdb-route`` and ``pouchdb-seamless-auth``.

I do think that whatever we decide, we should keep having the test suite depend upon them. ``pouchdb-rewrite``'s tests, which use both, are too valuable to lose, as I remember using them to cover a lot of edge cases for all kind of plugins.

**Comment by loic** [comment]
| Any idea why the rewrite tests were in there twice in the first place?

From my understanding, the `git subtree merge` of `pouchdb-rewrite` brought it to the monorepo, and it was copied instead of moved to the `tests` directory.

**Comment by marten-de-vries** [comment]
Ah, ok. So I've thought some more about this, and I think we should include ``pouchdb-seamless-auth`` into the monorepo as well. Its tests (which also cover ``pouchdb-auth``) are worth it, if nothing else. I can make a PR for it, if you want.

It does mean that I think we should revert the ``pouchdb-seamless-auth`` part of this PR.

**Comment by loic** [comment]
No worries, I'll take care of the merge.

**Comment by loic** [comment]
Removed the `seamless-auth` commit, so it should be ready for merging once CI has run.

**Comment by marten-de-vries** [comment]
Nice work.
